publication/src/main/scala/hmda/publication/reports/disclosure/D53.scala

```diff
@@ -0,0 +1,77 @@
+package hmda.publication.reports.disclosure
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import hmda.model.publication.reports.{ ApplicantIncome, Disposition, MSAReport }
+import hmda.publication.reports._
+import hmda.publication.reports.util.ReportUtil._
+import hmda.publication.reports.util.ReportsMetaDataLookup
+import hmda.query.model.filing.LoanApplicationRegisterQuery
+
+import scala.concurrent.Future
+
+case class D53(
+  respondentId: String,
+  institutionName: String,
+  year: Int,
+  msa: MSAReport,
+  applicantIncomes: List[ApplicantIncome],
+  total: List[Disposition],
+  reportDate: String = formattedCurrentDate,
+  table: String = D53.metaData.reportTable,
+  description: String = D53.metaData.description
+) extends DisclosureReport
+
+object D53 {
+  val metaData = ReportsMetaDataLookup.values("D53")
+  val dispositions = metaData.dispositions
+
+  // Table filters:
+  // Property Type: 1, 2
+  // Purpose of Loan: 3
+  def generate[ec: EC, mat: MAT, as: AS](
+    larSource: Source[LoanApplicationRegisterQuery, NotUsed],
+    fipsCode: Int,
+    respondentId: String,
+    institutionNameF: Future[String]
+  ): Future[D53] = {
+
+    val lars = larSource
+      .filter(lar => lar.respondentId == respondentId)
+      .filter(lar => lar.msa != "NA")
+      .filter(lar => lar.msa.toInt == fipsCode)
+      .filter { lar =>
+        (lar.loanType == 2 || lar.loanType == 3 || lar.loanType == 4) &&
```

Comment:
Are these the right filters for this report?

Reply:
😬 
Fixed in cfa1a80.